master/internal/command/command.go

```diff
@@ -419,24 +415,45 @@ func (c *command) toCommand(ctx *actor.Context) *commandv1.Command {
 		StartTime:    protoutils.ToTimestamp(ctx.Self().RegisteredTime()),
 		Username:     c.owner.Username,
 		ResourcePool: c.config.Resources.ResourcePool,
+		Config:       protoutils.ToStruct(c.config),
+		ExitStatus:   exitStatus,
 	}
 }
 
 func (c *command) toShell(ctx *actor.Context) *shellv1.Shell {
+	exitStatus := protoutils.DefaultStringValue
+	if c.exitStatus != nil {
+		exitStatus = *c.exitStatus
+	}
+
+	addresses := make([]*structpb.Struct, 0)
+	for _, addr := range c.addresses {
+		addresses = append(addresses, protoutils.ToStruct(addr))
+	}
+
 	return &shellv1.Shell{
-		Id:           ctx.Self().Address().Local(),
-		State:        c.State().Proto(),
-		Description:  c.config.Description,
-		StartTime:    protoutils.ToTimestamp(ctx.Self().RegisteredTime()),
-		Container:    c.container.Proto(),
-		PrivateKey:   c.metadata["privateKey"].(string),
-		PublicKey:    c.metadata["publicKey"].(string),
-		Username:     c.owner.Username,
-		ResourcePool: c.config.Resources.ResourcePool,
+		Id:             ctx.Self().Address().Local(),
+		State:          c.State().Proto(),
+		Description:    c.config.Description,
+		StartTime:      protoutils.ToTimestamp(ctx.Self().RegisteredTime()),
+		Container:      c.container.Proto(),
+		PrivateKey:     c.metadata["privateKey"].(string),
+		PublicKey:      c.metadata["publicKey"].(string),
+		Username:       c.owner.Username,
+		ResourcePool:   c.config.Resources.ResourcePool,
+		Config:         protoutils.ToStruct(c.config),
+		ExitStatus:     exitStatus,
+		Addresses:      addresses,
+		AgentUserGroup: protoutils.ToStruct(c.agentUserGroup),
 	}
 }
 
 func (c *command) toTensorboard(ctx *actor.Context) *tensorboardv1.Tensorboard {
+	exitStatus := protoutils.DefaultStringValue
```

Comment:
could we encode no status as `""` for our external facing apis? IMO it'd be more natural to parse than some string. wdy think